README.md

```diff
@@ -1,7 +1,334 @@
+    ModifiedDate = DateTime.UtcNow
+};
+
+DocumentPermissionSet adminSet = new DocumentPermissionSet {
+    AllowedPermissions = new List<PermissionIdentity> {
+        new PermissionIdentity(@"acme\admin", PermissionIdentityType.User)
+    }
+};
+
+DocumentPermissionLevel adminLevel = new DocumentPermissionLevel {
+    PermissionSets = new List<DocumentPermissionSet> { adminSet }
+};
+
+DocumentPermissionSet usersSet = new DocumentPermissionSet {
+    AllowedPermissions = new List<PermissionIdentity> {
+        new PermissionIdentity(@"acme\john", PermissionIdentityType.User)
+    },
+    DeniedPermissions = new List<PermissionIdentity> {
+        new PermissionIdentity(@"acme\cannotaccess", PermissionIdentityType.User)
+    }
+};
+
+DocumentPermissionLevel userLevel = new DocumentPermissionLevel {
+    PermissionSets = new List<DocumentPermissionSet> { usersSet }
+};
+
+verySecureDocument.Permissions.Add(adminLevel);
+verySecureDocument.Permissions.Add(userLevel);
+
+client.DocumentManager.AddOrUpdateDocument(sourceId, verySecureDocument, null);
+```
+
+Then, you can push your identities as we did before.
+```csharp
+PermissionIdentity allowedMember = new PermissionIdentity(@"acme\john", PermissionIdentityType.User);
+PermissionIdentity deniedMember = new PermissionIdentity(@"acme\cannotaccess", PermissionIdentityType.User);
+PermissionIdentity admin = new PermissionIdentity(@"acme\admin", PermissionIdentityType.User);
+
+PermissionIdentityBody bodyAllowedMember = new PermissionIdentityBody(allowedMember);
+bodyAllowedMember.Mappings.Add(new PermissionIdentity("john@acme.com", PermissionIdentityType.User));
+
+PermissionIdentityBody bodyDeniedMember = new PermissionIdentityBody(deniedMember);
+bodyDeniedMember.Mappings.Add(new PermissionIdentity("cannotaccess@acme.com", PermissionIdentityType.User));
+
+PermissionIdentityBody bodyAdmin = new PermissionIdentityBody(admin);
+bodyAdmin.Mappings.Add(new PermissionIdentity("admin@acme.com", PermissionIdentityType.User));
+
+List<PermissionIdentityBody> mappingsMembersToAddOrUpdate = new List<PermissionIdentityBody> {
+    bodyAllowedMember,
+    bodyDeniedMember,
+    bodyAdmin
+};
+
+// Push user mappings and group members in one batched operation.
+client.PermissionManager.AddOrUpdateIdentities(expandedProviderId, null, mappingsMembersToAddOrUpdate);
+```
+
+## Activate logging
+The SDK uses `log4net` as its logging library. It can be useful to have some logs in case of a problem. In your log4net configuration, add a logger with `Coveo` as the name. We use namespaces as logger name. Using `Coveo` as the logger name will get you all the logs from the SDK. 
+**Good to know:**
+* If you activate `Trace` level, you will get information about requests made by the SDK.
+
+## Explore!
+Feel free to explore the other `Managers` that the `ICoveoPlatformClient` provides to you. Don't hesitate to provide us feedback, report bugs if any, and ask for a feature!
```

Comment:
"to provide us feedback, report bugs if any, and ask for a feature!" ->
"to send us feedback, to report bugs if any, and to ask for a feature!"

Reply:
```suggestion
Feel free to explore the other `Managers` that the `ICoveoPlatformClient` provides to you. Don't hesitate to send us feedback, to report bugs if any, and to ask for a feature!
```